ext/search_kit/CRM/Search/BAO/SearchDisplay.php

```diff
@@ -16,6 +16,38 @@
  */
 class CRM_Search_BAO_SearchDisplay extends CRM_Search_DAO_SearchDisplay implements \Civi\Core\HookInterface {
 
+  /**
+   * Event fired after saving a SearchDisplay.
+   * @param \Civi\Core\Event\PostEvent $event
+   */
+  public static function self_hook_civicrm_post(\Civi\Core\Event\PostEvent $event) {
+    $display = $event->object;
+    $params = $event->params;
+    if (!$display->name || !$display->type || !$display->saved_search_id) {
+      $display->find(TRUE);
+    }
+    if ($display->type === 'autocomplete' && isset($params['is_autocomplete_default']) && (empty($params['check_permissions']) || CRM_Core_Permission::check('administer CiviCRM'))) {
```

Comment:
I think I would use `'administer CiviCRM data'` here rather than `administer CiviCRM` (the former is more granular but implied by the latter)

Reply:
@eileenmcnaughton it should match whatever permission is required by the **Settings - Search Preferences** form. I believe it's `administer CiviCRM` no?

Reply:
yeah - ok to leave